ext/filter/callback_filter.c

```diff
@@ -23,7 +23,7 @@ void php_filter_callback(PHP_INPUT_FILTER_PARAM_DECL)
 	int status;
 
 	if (!option_array || !zend_is_callable(option_array, 0, NULL)) {
-		php_error_docref(NULL, E_WARNING, "First argument is expected to be a valid callback");
+		zend_argument_type_error(1, "must be a valid callback");
```

Comment:
argument_type_error doesn't make sense here, it's not an argument.

Reply:
Fixed!